### PR TITLE
Regression test should not fail if required package is not available

### PR DESCRIPTION
**Pull request by praveenkuttappan**
If a required package is not available on PyPI, Regression Test currently fails. Test should exit gracefully if error is due to a missing required package on PyPI.

### CODE REVIEW

File: scripts/devops_tasks/common_tasks.py [hunk 1]
```diff
@@ -67,6 +69,7 @@
     "Regression": omit_regression,
 }
 
+lambda_filter_azure_pkg = lambda x: x.startswith("azure") and "-nspkg" not in x
```
Comment (scbedd) [inline scripts/devops_tasks/common_tasks.py]: Can you move this to the top?

File: scripts/devops_tasks/test_regression.py [hunk 2]
```diff
@@ -144,18 +148,19 @@ def _run_test(self, dep_pkg_path):
...
-        dep_pkg_name, _, _, _ = parse_setup(dep_pkg_path)
+        dep_pkg_name, version, _, _ = parse_setup(dep_pkg_path)
         release_tag = get_release_tag(dep_pkg_name, self.context.is_latest_depend_test)
         if not release_tag:
-            logging.error(
-                "Release tag is not avaiable. Skipping package {} from test".format(
-                    dep_pkg_name
-                )
-            )
+            logging.error("Release tag is not avaiable. Skipping package {} from test".format(dep_pkg_name))
```
Comment (scbedd) [inline scripts/devops_tasks/test_regression.py]: Nit: `avaiable` instead of `available`

**Comment by praveenkuttappan** [comment]
Successful run from playground pipeline: https://dev.azure.com/azure-sdk/playground/_build/results?buildId=257915&view=logs&j=3fadc984-4acf-5cfe-b896-a35ffcc3efec&t=3fadc984-4acf-5cfe-b896-a35ffcc3efec

**Comment by praveenkuttappan** [comment]
Successful run on internal pipeline for azure-core:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=257930&view=logs&j=3fadc984-4acf-5cfe-b896-a35ffcc3efec

**Pull request merged**